app/employee-app/backend/modules/database/db_queries.bal

```diff
@@ -0,0 +1,518 @@
+                        'office' , office_name,
+                        'location', office_location
+                        )
+                    )
+                FROM 
+                    hris_office o 
+                WHERE 
+                    o.company_id = c.company_id
+                    AND
+                    o.office_is_active = 1
+            ) as offices
+        FROM 
+            hris.hris_company c
+        WHERE 
+            c.company_is_active = 1
+    `;
+    CompanyFilter {ids, companies, locations} = filter;
+    sql:ParameterizedQuery[] filterQueries = [];
+    addFilterCondition(filterQueries, <sql:ParameterizedQuery>`c.company_id`, ids);
+    addFilterCondition(filterQueries, <sql:ParameterizedQuery>`c.company_name`, companies);
+    addFilterCondition(filterQueries, <sql:ParameterizedQuery>`c.company_location`, locations);
+    sqlQuery = buildSqlSelectQuery(sqlQuery, filterQueries);
+    sqlQuery = sql:queryConcat(sqlQuery, ` LIMIT ${'limit} OFFSET ${offset}`);
+
+    return sqlQuery;
+}
+
+# Build query to retrieve compensation data for given filter.
+#
+# + filter - Compensation filter
+# + return - sql:ParameterizedQuery - Select query for the compensation data
+isolated function getCompensationQuery(string filter) returns sql:ParameterizedQuery =>
+`
+    SELECT 
+        email_template_name as emailTemplate,
+	    email_template_key_value_pairs as keyValuePairs
+    FROM 
+        hris.hris_email_template
+    WHERE
+        email_template_name = ${filter}
+        AND
+        email_template_app_name = ${types:APP_NAME};
+`;
+
+# Build query to retrieve the employee from the active or marked leaver status and email.
+#
+# + email - Email of the employee
+# + return - sql:ParameterizedQuery - Select query for the employee table
+isolated function getEmployeeQuery(string email) returns sql:ParameterizedQuery {
+
+    sql:ParameterizedQuery mainQuery = getCommonEmployeeQuery();
+    sql:ParameterizedQuery finalQuery = sql:queryConcat(
+        mainQuery,
+        ` WHERE (e.employee_status = ${ACTIVE} OR e.employee_status = ${MARKED\ LEAVER}) AND e.employee_work_email = ${email}`
+    );
+    return finalQuery;
+}
+
+# Build query to retrieve the employees.
+#
+# + filters - Filter objects containing the filter criteria for the query
+# + 'limit - The maximum number of employees to return
+# + offset - The number of employees to skip before starting to collect the result set
+# + return - sql:ParameterizedQuery - Select query for the employee table
+isolated function getEmployeesQuery(EmployeeFilter filters, int 'limit, int offset)
+    returns sql:ParameterizedQuery {
+
+    sql:ParameterizedQuery mainQuery = getCommonEmployeeQuery();
+    EmployeeFilter {
+        status,
+        employmentType
+    } = filters;
+    sql:ParameterizedQuery[] filterQueries = [];
+    filterQueries.push(<sql:ParameterizedQuery>`SUBSTRING_INDEX(e.employee_work_email, '@', -1) = 
+        ${types:WSO2_DOMAIN}`);
```

Comment:
Also, there's unnecessary whitespace before the interpolation?

Reply:
Did that because it exceeds 120 characters.